### ISSUE TITLE
Document cargo features in readme

### ISSUE THREAD

**Issue by rukai** [jonhoo/inferno#299]
Users who use this crate as library will very likely want the cli feature disabled and its also not clear to me what the `nameattr` feature is for even after looking at the code.
So a section in the readme or maybe the root of the crate docs that lists what every feature is and which are enabled by default would be really valuable.

**Comment by jonhoo** [comment]
I completely agree! I really wish there was a standard way to document features in Cargo/the Rust ecosystem. There's some discussion in https://github.com/rust-lang/api-guidelines/discussions/43, but no hard guidance as of yet. @itamarst maybe you could take a stab at this after you land #294?
